library/src/main/java/com/google/android/exoplayer/extractor/ts/Mpeg2Reader.java

```diff
@@ -0,0 +1,189 @@
+package com.google.android.exoplayer.extractor.ts;
+
+import com.google.android.exoplayer.C;
+import com.google.android.exoplayer.MediaFormat;
+import com.google.android.exoplayer.extractor.TrackOutput;
+import com.google.android.exoplayer.util.Assertions;
+import com.google.android.exoplayer.util.MimeTypes;
+import com.google.android.exoplayer.util.ParsableByteArray;
+
+import android.util.Log;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Parses a continuous mpeg2 byte stream and extracts individual frames.
+ */
+/* package */ final class Mpeg2Reader extends ElementaryStreamReader {
+    private static final String TAG = "Mpeg2Reader";
+
+    private int flags;
+    private boolean seqHeaderFound;
+    private boolean frameCollected;
+    private boolean isSync;
+
+    private int width;
+    private int height;
+
+    private long timeUs;
```

Comment:
Can we use names consistent with the H264 reader? So specifically `sampleTimeUs` and `isKeyframe` rather than `timeUs` and `isSync`. Some grouping would also be nice, as in the H264 case, like:

```
  // State that should not be reset on seek.
  private boolean hasOutputFormat;

  // State that should be reset on seek.
  private boolean foundFirstSample;
  private long totalBytesWritten;

  // Per sample state that gets reset at the start of each sample.
  private boolean isKeyframe;
  private long samplePosition;
  private long sampleTimeUs;
```